### PR TITLE
Search: migrate manage dashboards

### LINKED ISSUE #23423: Migrate manage dashboards view to React
**What would you like to be added**:
Migrate the manage dashboards view at `/dashboards` to React.

**Why is this needed**:
Part of Search migration epic.

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:

Migrate manage dashboards to React.

Main changes (apart from component refactoring):
- Move search query and search handling into separate hooks. 
- Add `mergeReducers` function to compose multiple reducers with shared state structure.

**Which issue(s) this PR fixes**:

Fixes #23423

**Special notes for your reviewer**:
Next steps:
- Improve rendering performance by using virtualised list and possibly memoizing list components.
- Cleanup old angular files

### CODE REVIEW

File: public/app/features/manage-dashboards/partials/dashboard_list.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <page-header model="ctrl.navModel"></page-header>
 
 <div class="page-container page-body">
-  <manage-dashboards />
+  <!-- Need to pass the props, otherwise React component doesn't render -->
+  <manage-dashboards folder-id="undefined" folder-uid="undefined" />
```
Comment (torkelo) [inline public/app/features/manage-dashboards/partials/dashboard_list.html:5]: looks like we can make this a full react container instead now? or are there still things in the DashboardListCtrl  ?

Comment (Clarity-89) [reply public/app/features/manage-dashboards/partials/dashboard_list.html:5]: Good idea, do I need to add the component to the router using `<react-container />` for that?

File: public/app/core/angular_wrappers.ts [hunk 2]
```diff
@@ -88,6 +88,7 @@ export function registerAngularDirectives() {
     ['onTagFilterChange', { watchDepth: 'reference' }],
   ]);
   react2AngularDirective('searchWrapper', SearchWrapper, []);
+  react2AngularDirective('manageDashboards', ManageDashboards, ['folderId', 'folderUid']);
```
Comment (torkelo) [inline public/app/core/angular_wrappers.ts:91]: do we need this now?

Comment (Clarity-89) [reply public/app/core/angular_wrappers.ts:91]: Not anymore, forgot to remove this.

File: public/app/features/search/components/DashboardList.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import locationUtil from 'app/core/utils/location_util';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { ManageDashboards } from './ManageDashboards';
+
+interface Props {
+  navModel: NavModel;
+  uid?: string;
+}
+
+export const DashboardList: FC<Props> = ({ navModel, uid }) => {
```
Comment (torkelo) [inline public/app/features/search/components/DashboardList.tsx:19]: We usually name this top level container / page component with a Page suffix, so DashboardListPage (same for the filename)

Comment (torkelo) [reply public/app/features/search/components/DashboardList.tsx:19]: oh and use React.memo here so the whole page does not re-render on every redux state change

**Review by torkelo — APPROVED** [review]
Great job!  Created another search issue here: https://github.com/grafana/grafana/issues/23641 that I think would be easy and make search a better.

File: public/app/features/search/components/DashboardActions.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      url += `?folderId=${folderId}`;
+    }
+
+    return url;
+  };
+
+  return (
+    <HorizontalGroup spacing="md" align="center">
+      {canEdit && (
+        <a className="btn btn-primary" href={actionUrl('new')}>
```
Comment (tskarhed) [inline public/app/features/search/components/DashboardActions.tsx:24]: Use `LinkButton` here :)

Comment (Clarity-89) [reply public/app/features/search/components/DashboardActions.tsx:24]: Ah, nice! Let me know if there's any other form elements I need to update :)

**Pull request merged**